engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -250,8 +254,14 @@ private[spel] class Typer(classLoader: ClassLoader, commonSupertypeFinder: Commo
           }
         case _ => invalid("Invalid ternary operator") // shouldn't happen
       }
-      //TODO: what should be here?
-      case e: TypeReference => fixed(Unknown)
+
+      case e: TypeReference => {
+        (referenceTypeValidating, typeDefinitionSet.validateTypeReference(e)) match {
+          case (false, _) => valid(Unknown)
+          case (true, true) => valid(Unknown)
+          case _ => invalid("Class is not allowed to be passed as TypeReference")
```

Comment:
Can we move this invalid to `validateTypeReference`? `Validated` is better return type for method `validate`. Also validate is more aware about context of validation so is more close to message.

Reply:
I put some of the logic into validateMethod so it returns Validated but I'm not certain if that's exactly what you meant so I would appreciate it if you took another look at it

Reply:
You are right, i got a little fixated on that pattern matching there, it's much better this way. Thanks